modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/action/executor/test/ObjectActionExecutorRegistryImplTest.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.action.executor.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.object.action.executor.ObjectActionExecutor;
+import com.liferay.object.action.executor.ObjectActionExecutorRegistry;
+import com.liferay.object.scope.CompanyScoped;
+import com.liferay.object.scope.ObjectDefinitionScoped;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.UnicodeProperties;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+
+/**
+ * @author Guilherme Camacho
+ */
+@RunWith(Arquillian.class)
+public class ObjectActionExecutorRegistryImplTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testShouldNotReturnObjectActionExecutorScopedByAnotherCompany() {
+		ObjectActionExecutor testObjectActionExecutor1 =
+			new TestObjectActionExecutor(1, ListUtil.fromString("abc"));
+
+		ObjectActionExecutor testObjectActionExecutor2 =
+			new TestObjectActionExecutor(2, ListUtil.fromString("abc"));
+
+		ServiceRegistration<ObjectActionExecutor>
+			objectActionExecutorServiceRegistration1 = null;
+
+		ServiceRegistration<ObjectActionExecutor>
+			objectActionExecutorServiceRegistration2 = null;
+
+		try {
+			objectActionExecutorServiceRegistration1 =
+				_registerObjectActionExecutor(testObjectActionExecutor1);
+
+			objectActionExecutorServiceRegistration2 =
+				_registerObjectActionExecutor(testObjectActionExecutor2);
```

Comment:
@guilhermedcamacho we should declare variables closer than its use,

move testObjectActionExecutor2 to just before its use here,

and I believe we should rename it to testObjectActionExecutor since the first one was instantiated inline so we no longer have naming conflicts between these testObjectActionExecutors,

Thanks!